Nos/Models/NotificationViewModel.swift

```diff
@@ -0,0 +1,89 @@
+/// `loadContent()` to populate the `content` variable because it relies on some
+///  database queries.
+class NotificationViewModel: ObservableObject, Identifiable {
+    let noteID: HexadecimalString
+    let authorProfilePhotoURL: URL?
+    let actionText: AttributedString
+    @Published var content: AttributedString?
+    let date: Date
+    
+    var id: HexadecimalString {
+        noteID
+    }
+    
+    /// Generates a notification request that can be sent to the UNNotificationCenter to display a banner notification.
+    /// You probably want to call `loadContent(in:)` before accessing this.
+    var notificationCenterRequest: UNNotificationRequest {
+        let content = UNMutableNotificationContent()
+        content.title = String(actionText.characters)
+        if let attributedContent = self.content {
+            content.body = String(attributedContent.characters)
+        }
+        content.userInfo = ["eventID": id]
+        let trigger = UNTimeIntervalNotificationTrigger(timeInterval: 1, repeats: false)
+        return UNNotificationRequest(
+            identifier: noteID, 
+            content: content, 
+            trigger: trigger
+        )
+    }
+    
+    convenience init?(coreDataModel: NosNotification, context: NSManagedObjectContext) {
+        guard let eventID = coreDataModel.eventID,
+            let note = Event.find(by: eventID, context: context),
+            let user = coreDataModel.user else {
+            return nil
+        }
+        
+        self.init(note: note, user: user)
+    }
+    
+    init(note: Event, user: Author) {
+        self.noteID = note.identifier ?? ""
+        self.date = note.createdAt ?? .distantPast
+        self.authorProfilePhotoURL = note.author?.profilePhotoURL
+
+        // Compute action text
+        var actionText: AttributedString
+        var authorName = AttributedString("\(note.author?.safeName ?? Localized.someone.string) ")
+        var range = Range(uncheckedBounds: (authorName.startIndex, authorName.endIndex))
+        authorName[range].font = .boldSystemFont(ofSize: 17)
+        
+        if note.isReply(to: user) {
+            actionText = authorName + AttributedString(Localized.Reply.repliedToYourNote.string)
+        } else if note.references(author: user) {
+            actionText = authorName + AttributedString(Localized.Reply.mentionedYou.string)
+        } else {
+            actionText = AttributedString()
+        }
+        
+        range = Range(uncheckedBounds: (actionText.startIndex, actionText.endIndex))
+        actionText[range].foregroundColor = .primaryTxt
+        self.actionText = actionText
+        
+        content = AttributedString(note.content ?? "")
+    }
+    
+    /// Populates the `content` variable. This is not done at init in order to keep
+    /// it synchronous for use in a View.
+    @MainActor @discardableResult func loadContent(in context: NSManagedObjectContext) async -> AttributedString? {
```

Comment:
I know this function is kludgy but I spent too much time trying to find a nice way to do it and decided it wasn't worth it in the end. 

Probably the right way to do this is create a custom datasource wrapper around the @FetchRequest but that would be some tricky work I think.